### PR TITLE
Use new output environment files

### PR DESCRIPTION
**Pull request by matthewrees-latacora**
GitHub have deprecated the old `set-output` method in favour of environment files, so this change should remove the deprecation warning when using this reusable action. More info: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

### CODE REVIEW

**Review by ghost — APPROVED** [review]
LGTM.

Do you think maybe we should test this before we merge it? When I reference actions in workflow files, I tend to use either very specific tags or commit SHAs. But it’s certainly possible that some of our workflows don’t specify a ref, or specify `main`.

OTOH… not specifying a ref, or specifying `main`, is a bad idea, and if this change breaks such a workflow, that might actually be not so terrible; it might motivate us to fix the unstable references…

So yeah, I’d be comfortable either way. Your call.

**Comment by matthewrees-latacora** [comment]
I'm happy to merge this here and then test this revision hash in the sk8soc repo to see that it works correctly and report back!

Edit: To clarify, every occurrence of using this workflow that I've seen has used commit hashes, so I'm quite sure we won't break anything by merging this untested, but as you said, if we do it would be a great signal to fix that.

**Comment by matthewrees-latacora** [comment]
Looks okay to me: https://github.com/latacora/skateboard-soc/actions/runs/3963921910

There is no deprecation warning from the version workflow, and the workflow summary has version numbers for the build artifacts, so it looks 💯 

Yup:
`tag="${{ needs.version.outputs.version_id }}"`
turned into
`tag="20230119-gha-output-deprecation-f622b3e"`

**Comment by ghost** [comment]
🎉

**Pull request merged**
